### PR TITLE
[NewUI] New account modal allows for addition of account name.

### PR DESCRIPTION
**Pull request by danjm**
This PR enables the user to successfully give a name to an account created through the new account modal.

newaccountname

### CODE REVIEW

File: ui/app/actions.js [hunk 1]
```diff
@@ -348,9 +348,25 @@ function navigateToNewAccountScreen () {
   }
 }
 
-function addNewAccount () {
+function addNewAccount (oldIdentities) {
```
Comment (chikeichan) [inline ui/app/actions.js]: we can get this using `getState` instead of relying on caller of the action to pass it in

File: ui/app/actions.js [hunk 2]
```diff
@@ -348,9 +348,25 @@ function navigateToNewAccountScreen () {
...
+      background.addNewAccount((err, { identities: newIdentities}) => {
+        if (err) {
+          dispatch(actions.displayWarning(err.message))
+          return reject(err)
+        }
+        const newAccountAddress = Object.keys(newIdentities).find(address => !oldIdentities[address])
+
+        dispatch(actions.hideLoadingIndication())
+
+        forceUpdateMetamaskState(dispatch)
```
Comment (chikeichan) [inline ui/app/actions.js]: why is this necessary?

File: ui/app/components/modals/new-account-modal.js [hunk 3]
```diff
@@ -19,21 +20,31 @@ function mapDispatchToProps (dispatch) {
     hideModal: () => {
       dispatch(actions.hideModal())
     },
-    createAccount: () => {
-      dispatch(actions.addNewAccount())
-      dispatch(actions.hideModal())
+    createAccount: (identities, newAccountName) => {
+      dispatch(actions.addNewAccount(identities))
+        .then((newAccountAddress) => {
```
Comment (chikeichan) [inline ui/app/components/modals/new-account-modal.js]: Do we still need to save account label if user didn't enter a label?

File: ui/app/actions.js (hunk 2 above)
Comment (danjm) [reply ui/app/actions.js]: This is in the `callBackgroundThenUpdate` fubction. I replaced the `callBackgroundThenUpdate(ackground.addNewAccount)`  call with the code above, and didn't want to remove the `forceUpdateMetamaskState()` call.

I don't know exactly why it is necessary, but has to with maintaining the correct client side state when the background state changes.

File: ui/app/actions.js (hunk 1 above)
Comment (danjm) [reply ui/app/actions.js]: Done in 69682a3f

File: ui/app/components/modals/new-account-modal.js (hunk 3 above)
Comment (danjm) [reply ui/app/components/modals/new-account-modal.js]: No. Fixed in 69682a3f
